pkg/provider/oracle/oracle.go

```diff
@@ -394,9 +378,45 @@ func (vms *VaultManagementService) ValidateStore(store esv1beta1.GenericStore) e
 		return err
 	}
 
+	if oracleSpec.ServiceAccountRef != nil {
+		if err := utils.ValidateReferentServiceAccountSelector(store, *oracleSpec.ServiceAccountRef); err != nil {
+			return fmt.Errorf("invalid ServiceAccountRef: %w", err)
+		}
+	}
+
 	return nil
 }
 
+func (vms *VaultManagementService) getWorkloadIdentityProvider(serviceAcccountRef *esmeta.ServiceAccountSelector, region, namespace string) (common.ConfigurationProvider, error) {
+	defer func() {
+		_ = os.Unsetenv(auth.ResourcePrincipalVersionEnvVar)
+		_ = os.Unsetenv(auth.ResourcePrincipalRegionEnvVar)
+		vms.workloadIdentityMutex.Unlock()
+	}()
```

Comment:
```suggestion
func (vms *VaultManagementService) getWorkloadIdentityProvider(serviceAcccountRef *esmeta.ServiceAccountSelector, region, namespace string) (_ common.ConfigurationProvider, err error) {
	defer func() {
		if uerr := os.Unsetenv(auth.ResourcePrincipalVersionEnvVar); uerr != nil {
			err = errors.Join(err, fmt.Errorf("unable to set OCI SDK environment variable %s: %w", auth.ResourcePrincipalRegionEnvVar, uerr))
		}

		if uerr := os.Unsetenv(auth.ResourcePrincipalRegionEnvVar); uerr != nil {
			err = errors.Join(err, fmt.Errorf("unabled to unset OCI SDK environment variable %s: %w", auth.ResourcePrincipalVersionEnvVar, uerr))
		}

		vms.workloadIdentityMutex.Unlock()
	}()
```

Let's not ignore those unsets? :)

Reply:
Sounds good